field/TinyLang/Field/Generator.hs

```diff
@@ -278,79 +255,44 @@ arbitraryBinOpRing = elements [Add, Sub, Mul]
-                 reduce the risk of this, or supply a separate list of
-                 variables which we're certain will only contain integer
-                 values.
-               -}
-              (2, EIf <$>
-                boundedArbitraryExprB vars (size `Prelude.div` 3) <*>
-                boundedArbitraryExprI vars (size `Prelude.div` 3) <*>
-                boundedArbitraryExprI vars (size `Prelude.div` 3)),
-              (2, do
-                 uniVar <- genFreshUniVar
-                 let vars' = Some uniVar : vars
-                 ELet uniVar <$>
-                      (boundedArbitraryExprI vars  (size `Prelude.div` 2)) <*>
-                      (boundedArbitraryExprI vars' (size `Prelude.div` 2))
-              ),
-              (2, do
-                 uniVar <- genFreshUniVar
-                 let vars' = Some uniVar : vars
-                 ELet uniVar <$>
-                      (boundedArbitraryExprB vars  (size `Prelude.div` 2)) <*>
-                      (boundedArbitraryExprI vars' (size `Prelude.div` 2))
-              ),
-              (2, EAppUnOp <$> arbitraryUnOpRing <*> boundedArbitraryExprI vars (size-1)),
-              (2, EAppBinOp <$>
-                arbitraryBinOpRing <*>
-                boundedArbitraryExprI vars (size `Prelude.div` 2) <*>
-                boundedArbitraryExprI vars (size `Prelude.div` 2))
-             ]
-
--- | Either a general field expression or one which is guaranteed to
--- be integer-valued.  We want the latter most of the time, but not
--- always.
-usuallyIntValuedArbitraryExpr ::
-    (Field f, Arbitrary f, MonadGen m, MonadSupply m)
-    => Vars f -> Int -> m (Expr f (AField f))
-usuallyIntValuedArbitraryExpr vars size =
-    frequency [(1, boundedArbitraryExprF vars size),
-               (9, boundedArbitraryExprI vars size)
-         ]
-
--- Generate an expression from a collection of variables with the
--- number of nodes (approximately) bounded by 'size'
-boundedArbitraryExpr
-    :: (Field f, Arbitrary f, MonadGen m, MonadSupply m)
-    => Vars f -> Int -> m (SomeUniExpr f)
-boundedArbitraryExpr vars size =
-    do
-      adjustUniquesForVars vars
-      oneof [SomeUniExpr Bool <$> boundedArbitraryExprB vars size,
-             SomeUniExpr Field <$> boundedArbitraryExprF vars size]
-
--- Generate an expression over the default variables.  Again, this is bounded by 'size'.
-defaultArbitraryExpr
-    :: (Field f, Arbitrary f, MonadGen m, MonadSupply m)
-    => Int -> m (SomeUniExpr f)
-defaultArbitraryExpr = boundedArbitraryExpr defaultVars
-
--- We can shrink any expression to just a hardcoded ground value (except we shouldn't shrink other
--- ground values to hardcoded ground values to prevent looping).
-defaultUniVal :: forall f a. (KnownUni f a, Field f) => UniVal f a
-defaultUniVal = case knownUni @f @a of
-    Bool  -> UniVal Bool True
-    Field -> UniVal Field $ fromInteger 100
+boundedArbitraryExprI _    size | size <= 1 = EVal <$> arbitraryValI
```

Comment:
In contrast with boundedArbitraryExpr, we really don't want EVar here (for the same reason it's not included just below)